src/main/java/xfacthd/framedblocks/common/compat/jei/camo/CamoItemStackHelper.java

```diff
@@ -0,0 +1,83 @@
+package xfacthd.framedblocks.common.compat.jei.camo;
+
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.item.ItemStack;
+import org.jetbrains.annotations.Nullable;
+import xfacthd.framedblocks.api.block.IFramedBlock;
+import xfacthd.framedblocks.api.camo.CamoContainer;
+import xfacthd.framedblocks.api.camo.CamoContainerFactory;
+import xfacthd.framedblocks.api.camo.CamoContainerHelper;
+import xfacthd.framedblocks.api.util.CamoList;
+import xfacthd.framedblocks.common.FBContent;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class CamoItemStackHelper
+{
+    private CamoItemStackHelper()
+    {
+    }
+
+
+    @Nullable
+    public static CamoContainerFactory<?> getCamoContainerFactory(ItemStack itemStack)
+    {
+        CamoContainerFactory<?> factory = CamoContainerHelper.findCamoFactory(itemStack);
+        if (factory == null || !factory.canApplyInCraftingRecipe(itemStack))
+        {
+            return null;
+        }
+        return factory;
+    }
+
+    @Nullable
+    public static IFramedBlock getFramedBlock(ItemStack itemStack)
+    {
+        if (itemStack.getItem() instanceof BlockItem item && item.getBlock() instanceof IFramedBlock framedBlock)
+        {
+            return framedBlock;
+        }
+        return null;
+    }
+
+    public static boolean isDoubleFramedBlock(ItemStack itemStack)
+    {
+        IFramedBlock framedBlock = getFramedBlock(itemStack);
+        return framedBlock != null && isDoubleFramedBlock(framedBlock);
+    }
+
+    public static boolean isDoubleFramedBlock(IFramedBlock framedBlock)
+    {
+        return framedBlock.getBlockType().consumesTwoCamosInCamoApplicationRecipe();
+    }
+
+    public static boolean isEmptyFramedBlock(ItemStack itemStack)
+    {
+        IFramedBlock framedBlock = getFramedBlock(itemStack);
+        if (framedBlock == null)
+        {
+            return false;
+        }
+        CamoList camos = itemStack.get(FBContent.DC_TYPE_CAMO_LIST);
+        return camos == null || camos.isEmpty();
```

Comment:
I added a method for checking whether the list is empty or all of its contents are considered empty. Please rebase onto the latest commits and use that method as it avoids issues when a player manages to acquire a framed block where the list is present and non-empty but only contains empty camos.
```suggestion
        return camos == null || camos.isEmptyOrContentsEmpty();
```